Zif 970

giuliano176 opened the pull request:
REF: ZIF-970

This PR enables the user to select some attachments for smart link conversion.
Once selected, the send action will call the createSmartLinks API and remove the converted attachments.
The public URL generated by the API call will be injected in the body of the message, after the signature.

The feature works with:

- send from editor
- send from draft
- send later option

FrancescoGottardi committed:
wip: generate public link from attachment

giuliano176 committed:
WIP: remove folder selection

giuliano176 committed:
WIP: add remove attachment function

giuliano176 committed:
WIP: clean up unused functions

FrancescoGottardi committed:
feat: flag attachment to convert in smartLink

AlessioCoser committed:
refactor: cleanup a bit the convertToSmartLink store function

giuliano176 committed:
feat: change color of smart links

zextrasGiovanniDefacci committed:
chore: introduce local store for saving smartlinks info

FrancescoGottardi committed:
test: update attachment preview test to handle localStorage state

AlessioCoser committed:
WIP: update smartLink state on save and load draft

giuliano176 committed:
feat: persist smartLinks when opening a saved draft

FrancescoGottardi committed:
wip: SendMsgWithSmartLink soap

giuliano176 committed:
WIP: enable toggle of smartLinks in local storage

FrancescoGottardi committed:
fix: correct SendMsgWithSmartLinks soap name

FrancescoGottardi committed:
wip: refactor

giuliano176 committed:
WIP: add createSmartLinks button and api call

AlessioCoser committed:
feat: fix createSmartLink request field

giuliano176 committed:
chore: graphical changes to convert to smart link button

giuliano176 committed:
feat: use create smart link api

giuliano176 committed:
fix: fix fetch of create smart link by sending only the links related to the current draft

giuliano176 committed:
feat: added hidden property to smart links html element

giuliano176 committed:
fix: restore sendMsg on sending of message action

giuliano176 committed:
test: skip edit view test

giuliano176 committed:
chore: remove hidden attribute in smart link html element

FrancescoGottardi committed:
feat: moved smartLinks generation at the send action

giuliano176 committed:
feat: disable conversion to smart links when there aren't attachments to convert

giuliano176 committed:
feat: toggle mark as smart link

giuliano176 committed:
fix: after the conversion remove attachments that are marked as smartLinks

FrancescoGottardi committed:
feat: improved smartLink preview design

giuliano176 committed:
feat: added animated loader to send button

giuliano176 committed:
WIP: add create smart links on send later button

FrancescoGottardi committed:
fix: removing the correct attachments from the storage

FrancescoGottardi committed:
refactor: draftSmartLinks filtering for editView

giuliano176 committed:
feat: added animatedLoader to send multiButton

giuliano176 committed:
chore: restore closeModal on scheduled send

giuliano176 committed:
feat: remove smartLink from local storage when the attachment is removed

giuliano176 committed:
chore: clean up

FrancescoGottardi committed:
feat: show filename of the smarlink inside the mail body

giuliano176 committed:
chore: update commons

FrancescoGottardi committed:
refactor: extract generateSmartLinkHtml function

FrancescoGottardi committed:
Merge branch 'ZIF-970' of github.com:zextras/carbonio-mails-ui into ZIF-970

FrancescoGottardi committed:
refactor: extract generateSmartLinkHtml function

FrancescoGottardi committed:
fix: moved paddding inside attachmentPreview

giuliano176 committed:
chore: update commons

giuliano176 committed:
feat: remove smart link from local storage on sendMsg and sendMsgFromEditor

giuliano176 committed:
refactor: improve handling of smartlinks local storage

giuliano176 committed:
test: fix edit view test

giuliano176 committed:
feat: handle smart links in draft preview panel

giuliano176 committed:
feat: clean up smart links local storage on deleteEditor action

FrancescoGottardi committed:
feat: persist smartlink statuses across devices

giuliano176 committed:
fix: complete removal of local storage for managing smart links

giuliano176 committed:
chore: add requiresSmartLinkConversion attribute to saveDraft

AlessioCoser committed:
fix: fix typo when reading the requiresSmartLinkConversion attribute on attachment-block inside the detail-panel

giuliano176 committed:
Merge branch 'devel' into ZIF-970

giuliano176 committed:
WIP: keyboard change

giuliano176 committed:
WIP: change keyboard

FrancescoGottardi committed:
wip: refactor getSendDratAction

AlessioCoser committed:
fix: generate an editor on send button click

This is because we want to reuse the same methods to convert the attachment to smartLinks and send the email of the composer flow.

giuliano176 committed:
feat: send from draft works with smart link attachments

giuliano176 committed:
chore: clean up code

giuliano176 committed:
test: added test for generateSmartLinkHtml function

FrancescoGottardi committed:
test: added tests to  addSmartLinksToText

FrancescoGottardi committed:
wip: keyboard change

giuliano176 committed:
fix: fix generateSmartLinkHTML function

giuliano176 committed:
test: add tests for createSmartLink function and related

giuliano176 committed:
chore: fix icon restore attachment

giuliano176 committed:
chore: fix conflicts with devel

giuliano176 committed:
chore: clean up code

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/zextras/carbonio-mails-ui?pullRequest=560) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/zextras/carbonio-mails-ui?pullRequest=560) before we can accept your contribution.<br/>**2** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: Gammax92<br/>:white_check_mark: giuliano176<br/>:x: zextrasGiovanniDefacci<br/>:x: AlessioCoser<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/zextras/carbonio-mails-ui?pullRequest=560) it.</sub>

giuliano176 committed:
Merge branch 'devel' into ZIF-970

giuliano176 committed:
refactor: improve generateSmartLinksHtml and addSmartLinksToText functions

giuliano176 committed:
refactor: rename and createSmartLinks to updateEditorWithSmartLinks and extract API call

giuliano176 committed:
fix: remove bottom border on attachments

giuliano176 committed:
fix: reverse Refresh icon diretion for smart links

AlessioCoser committed:
feat: add correct convert to smart link button icon

AlessioCoser committed:
fix: fix border botton when attachment is not a smart link

nubsthead commented on src/ui-actions/message-actions.tsx:
```diff
@@ -500,8 +500,7 @@ export function replyMsg({
-	id,
-	folderId
+	id
 }: Pick<MessageActionPropType, 'id' | 'folderId'>): MessageActionReturnType {
```
All the types below must be updated since you removed folderId from params

giuliano176 replied:
done, thank you

nubsthead commented on src/ui-actions/message-actions.tsx:
```diff
@@ -520,8 +519,7 @@ export function replyAllMsg({
-	id,
-	folderId
+	id
 }: Pick<MessageActionPropType, 'id' | 'folderId'>): MessageActionReturnType {
```
here

giuliano176 replied:
done, thank you

nubsthead commented on src/ui-actions/message-actions.tsx:
```diff
@@ -540,8 +538,7 @@ export function forwardMsg({
-	id,
-	folderId
+	id
 }: Pick<MessageActionPropType, 'id' | 'folderId'>): MessageActionReturnType {
```
here

giuliano176 replied:
done, thank you

nubsthead commented on src/ui-actions/message-actions.tsx:
```diff
@@ -561,7 +558,6 @@ export function editAsNewMsg({
 	id,
-	folderId,
 	message
 }: Pick<MessageActionPropType, 'id' | 'folderId' | 'message'>): MessageActionReturnType {
```
here

giuliano176 replied:
done, thank you

nubsthead commented on src/views/app/detail-panel/edit/parts/edit-view-send-buttons.tsx:
```diff
@@ -48,7 +51,7 @@ export const EditViewSendButtons: FC<EditViewSendButtonsProps> = ({
 
 	const isSendLaterAllowed = useMemo(
-		() => attrs?.zimbraFeatureMailSendLaterEnabled === 'TRUE',
+		() => attrs?.zimbraFeatureMailSendLaterEnabled === 'TRUE' || true,
```
is the final true the default behaviour?

nubsthead commented on src/views/app/detail-panel/edit/parts/edit-view-send-buttons.tsx:
```diff
@@ -78,6 +81,10 @@ export const EditViewSendButtons: FC<EditViewSendButtonsProps> = ({
 						onClick={onSendNow}
 						disabledPrimary={disabled}
 						disabledSecondary={disabled}
+						// TODO: remove this ts-ignore once the type is fixed in the DS
```
Do you have an issue reference to keep track of this?

giuliano176 replied:
I see it has already been addressed on the devel branch, i assume the next shell will bring in these changes

giuliano176 committed:
fix: fix border botton when attachment is not a smart link

AlessioCoser committed:
refactor: WIP add a promise rejection to updateEditorWithSmartLinks

danielemegna committed:
feat: set proper type for CreateSmartLinksResponse when an error occours

danielemegna committed:
feat: catch createSmartLinksAction errors preventing email send

catch errors coming from createSmartLinksAction and invoke onSendError
to open snackbar with error message and reopen the mail editor

anticipate close editor action in order to
prevent additional editor opening on error

danielemegna committed:
style: make eslit happy :nail_care:

danielemegna committed:
test: improve incomplete updateEditorWithSmartLinks error scenario test

AlessioCoser committed:
lint: add a lint-fix command to help fix the linting automatically

AlessioCoser committed:
test: improve CreateSmartLink failure on edit-view component test

giuliano176 committed:
feat: improve CreateSmartLink failure on send from draft

giuliano176 committed:
refactor: improve sendDraftFromPreview with promise chain

giuliano176 committed:
feat: handle createSmartLink error in send later

FrancescoGottardi committed:
style: improved smart link url render inside the mail

FrancescoGottardi committed:
style: truncating the smartlink filename

giuliano176 committed:
chore: clean up code

giuliano176 committed:
chore: added sonar project properties

giuliano176 committed:
chore: merge devel into branch

giuliano176 committed:
refactor: remove unused folderId props from message-actions

giuliano176 committed:
chore: clean up code

giuliano176 committed:
fix: on sendFromDraft convert to smart link only if required

giuliano176 committed:
chore: clean up code

giuliano176 committed:
refactor: reduce complexity of useMessageActions function

giuliano176 committed:
refactor(type): improve ActionGeneratorProps definition

giuliano176 committed:
chore: clean up code

giuliano176 committed:
refactor: improve getSendDraftAction function

giuliano176 committed:
chore: update jest config options to generate lconv coverage report

giuliano176 committed:
test(build): add test for createSmartLink API call failure

giuliano176 committed:
chore: add html report coverage to jest config

giuliano176 committed:
test(build): add tests for editor transformation functions

FrancescoGottardi committed:
test: added getReferredContentIds case tests

FrancescoGottardi committed:
test: added editor toggleSmartLink case tests

giuliano176 committed:
Merge branch 'devel' into ZIF-970

giuliano176 committed:
chore: update commons submodule
